pkg/webhook/workloadspread/validating/workloadspread_validation.go

```diff
@@ -0,0 +1,253 @@
+			} else {
+				if subset.RequiredNodeSelectorTerm != nil {
+					coreNodeSelectorTerm := &core.NodeSelectorTerm{}
+					if err := corev1.Convert_v1_NodeSelectorTerm_To_core_NodeSelectorTerm(subset.RequiredNodeSelectorTerm.DeepCopy(), coreNodeSelectorTerm, nil); err != nil {
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("requiredNodeSelectorTerm"), subset.RequiredNodeSelectorTerm, fmt.Sprintf("Convert_v1_NodeSelectorTerm_To_core_NodeSelectorTerm failed: %v", err)))
+					} else {
+						allErrs = append(allErrs, corevalidation.ValidateNodeSelectorTerm(*coreNodeSelectorTerm, fldPath.Index(i).Child("requiredNodeSelectorTerm"))...)
+					}
+				}
+
+				if subset.PreferredNodeSelectorTerms != nil {
+					corePreferredSchedulingTerms := make([]core.PreferredSchedulingTerm, 0, len(subset.PreferredNodeSelectorTerms))
+					for i, term := range subset.PreferredNodeSelectorTerms {
+						corePreferredSchedulingTerm := &core.PreferredSchedulingTerm{}
+						if err := corev1.Convert_v1_PreferredSchedulingTerm_To_core_PreferredSchedulingTerm(term.DeepCopy(), corePreferredSchedulingTerm, nil); err != nil {
+							allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("preferredSchedulingTerms"), subset.PreferredNodeSelectorTerms, fmt.Sprintf("Convert_v1_PreferredSchedulingTerm_To_core_PreferredSchedulingTerm failed: %v", err)))
+						} else {
+							corePreferredSchedulingTerms = append(corePreferredSchedulingTerms, *corePreferredSchedulingTerm)
+						}
+					}
+
+					allErrs = append(allErrs, corevalidation.ValidatePreferredSchedulingTerms(corePreferredSchedulingTerms, fldPath.Index(i).Child("preferredSchedulingTerms"))...)
+				}
+			}
+
+			if subset.Tolerations != nil {
+				var coreTolerations []core.Toleration
+				for i, toleration := range subset.Tolerations {
+					coreToleration := &core.Toleration{}
+					if err := corev1.Convert_v1_Toleration_To_core_Toleration(&toleration, coreToleration, nil); err != nil {
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("tolerations"), subset.Tolerations, fmt.Sprintf("Convert_v1_Toleration_To_core_Toleration failed: %v", err)))
+					} else {
+						coreTolerations = append(coreTolerations, *coreToleration)
+					}
+				}
+				allErrs = append(allErrs, corevalidation.ValidateTolerations(coreTolerations, fldPath.Index(i).Child("tolerations"))...)
+			}
+
+			if subset.MaxReplicas != nil {
+				if i < len(subsets)-1 {
+					subsetMaxReplicas, err := intstr.GetValueFromIntOrPercent(subset.MaxReplicas, 100, true)
+					if err != nil || subsetMaxReplicas < 0 {
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("maxReplicas"), subset.MaxReplicas, "maxReplicas is not valid for subset"))
+					}
+					if subset.MaxReplicas.Type == intstr.String && subsetMaxReplicas > 100 {
```

Comment:
1. do we need to check whether the total maxreplicas is less than 1 if all maxreplicas is specified as percent
2. what if the total maxreplicas is greater than the workload replicas?

Reply:
1. Less than 1 should be permitted because the last subset's maxReplicas is nil, which will contain the remaining Pods.
2. We can validate the **percent** condition if the total maxReplicas > workload Replicas. The other condition should be valid because we can't know the actual workload replicas when validating.